Source/HTTPHeaders.swift

```diff
@@ -24,40 +24,301 @@
 
 import Foundation
 
-public typealias HTTPHeaders = [String: String]
 
-extension Dictionary where Key == String, Value == String {
-    public static func authorization(username: String, password: String) -> HTTPHeaders {
-        let credential = Data("\(username):\(password)".utf8).base64EncodedString()
+/// An order-preserving and case-insensitive representation of HTTP headers.
+public struct HTTPHeaders {
+    private var headers = [HTTPHeader]()
+
+    /// Create an empty instance.
+    public init() { }
+
+    /// Create an instance from an array of `HTTPHeader`s. Duplicate case-insensitive names are collapsed into the last
+    /// name and value encountered.
+    public init(_ headers: [HTTPHeader]) {
+        self.init()
+
+        headers.forEach { update($0) }
+    }
+
+    /// Create an instance from a `[String: String]`. Duplicate case-insensitive names are collapsed into the last name
+    /// and value encountered.
+    public init(_ dictionary: [String: String]) {
+        self.init()
+
+        dictionary.forEach { update(HTTPHeader(name: $0.key, value: $0.value)) }
+    }
+
```

Comment:
I can't help but think the `HTTPHeaders` type is missing an `add(_ header: HTTPHeader)` API. I get that `update` can be used for both, but it almost seems like an `add` API makes sense as well for completion sake that could simply call `update` internally. Thoughts? I'm sure you've already rolled this around...